### PR TITLE
Modify Body validity condition and add tests for Title and Body

### PR DESCRIPTION
**Pull request by firzanarmani**
`isValidBody()` was not updated in previous `Body` implementation to allow for `""`.

`""` allows for initialization of blocks without any Body content given by user, i.e. root block or blocks with only Title given by user.

However, a body containing only whitespace is invalid, such as `" "` or `"   "`. 

In this PR, test cases have also been written for `Body` and `Title` classes.

### CODE REVIEW

File: src/test/java/com/notably/model/block/BodyTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        // Valid body
+        assertTrue(Body.isValidBody("")); // Empty body (for Body initialization without content)
+        assertTrue(Body.isValidBody("25110048")); // Numbers only
+        assertTrue(Body.isValidBody("lorem ipsum dolor sit amet, consectetur adipiscing elit")); // Letters only
+        assertTrue(Body.isValidBody("lecture 2 begins with 4 examples")); // Alphanumeric characters
+        assertTrue(Body.isValidBody("Lecture Week 2")); // With uppercase characters
+        assertTrue(Body.isValidBody("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.")); // With non-alphanumeric characters
+        assertTrue(Body.isValidBody("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.")); // Long body
```
Comment (kevinputera) [inline src/test/java/com/notably/model/block/BodyTest.java]: Actually, I'm not exactly sure whether or not this (having so many tests on very similar cases) is a good idea. In a way, you can just add the include the last `assertTrue` test (perhaps also add in some numbers to the long body) and similar scenarios would have been tested.

File: src/main/java/com/notably/model/block/Body.java [hunk 2]
```diff
@@ -29,7 +35,7 @@ public Body(String body) {
      * @param test The body string to test.
      */
     public static boolean isValidBody(String test) {
-        return !test.isBlank();
+        return !test.matches(VALIDATION_REGEX);
```
Comment (kevinputera) [inline src/main/java/com/notably/model/block/Body.java]: Wouldn't it seem rather counter-intuitive that the `VALIDATION_REGEX` here actually matches the invalid scenario? It sort of breaks the convention of the other classes, where `VALIDATION_REGEX` always contains the regex that verifies for correctness.

**Review by kevinputera — APPROVED** [review]
LGTM, some comments below

File: src/main/java/com/notably/model/block/Body.java (hunk 2 above)
Comment (firzanarmani) [reply src/main/java/com/notably/model/block/Body.java]: Understood! Will modify the `VALIDATION_REGEX`

File: src/test/java/com/notably/model/block/BodyTest.java (hunk 1 above)
Comment (firzanarmani) [reply src/test/java/com/notably/model/block/BodyTest.java]: I've attempted to keep it a little simpler!

**Pull request merged**